pkg/agent/flowexporter/flowrecords/flow_records.go

```diff
@@ -58,6 +58,10 @@ func (fr *FlowRecords) AddOrUpdateFlowRecord(key flowexporter.ConnectionKey, con
 			IsActive:           true,
 		}
 	} else {
+		// If the connection is in dying state there is no need to update the record again
+		if flowexporter.IsConnectionDying(&conn) {
+			return nil
+		}
```

Comment:
Is this the correct place for this check? We already delete the record from the flow record map and record doesn't exist anymore.
Please add a unit test for optimization scenario if possible.

Reply:
Use case here is to avoid updating existing records when the connection just enters dying state. These records will be removed afterwards.
Sure. working on adding some unit tests and fixing e2e tests.

Reply:
Even after removing the records, for every connection in the connection map we call this routine (L109 connections.go). Therefore, we should move these to the beginning of function before checking if record exists or not.

Reply:
That makes sense. I added another function to set flow records to nil when it is exported and corresponding connection state is dying to avoid the case when the connection comes in as dying state we cannot add and export records.